build/build.rs

```diff
@@ -5,7 +5,10 @@ use std::env;
 use std::iter::once;
 use std::path::PathBuf;
 
-use ::bindgen::callbacks::{IntKind, ParseCallbacks};
+use ::bindgen::{
```

Comment:
How would that affect `esp-idf-hal` and `esp-idf-svc`? Would these continue to compile just fine after these changes, or do we need a PR for each of those to fix any compilation errors? 

Also: why do we want Rust enums?

Reply:
> How would that affect `esp-idf-hal` and `esp-idf-svc`? 

I guess there will need to be some changes to those crates.

> Also: why do we want Rust enums?

They are more ergonomic to use and the compiler will check if all variants are handled.

Reply:
> > How would that affect `esp-idf-hal` and `esp-idf-svc`?
> 
> I guess there will need to be some changes to those crates. I can have a look at that of course.

Would appreciate that. :)

> > Also: why do we want Rust enums?
> 
> They are more ergonomic to use and the compiler will check if all variants are handled.

I *think* I tried this once, but then for some reason abandoned it. Perhaps once you try to adjust `esp-idf-svc` and `esp-idf-hal` to work with the new enum-based constants, the problem (if there was a real one) will resurface itself.